Add basic marker on shape tests.

Tavmjong opened the pull request:
Basic marker on shape tests.

<!-- Reviewable:start -->

<!-- Reviewable:end -->

Tavmjong committed:
Add basic marker on shape tests.

svgeesus commented on svg/painting/reftests/marker-shape-001-ref.svg:
```diff
@@ -0,0 +1,59 @@
+          title="NAME_OF_REVIEWER"
+          href="mailto:EMAIL OR http://CONTACT_PAGE" />
+          <!-- YYYY-MM-DD -->
+    <html:link rel="help"
```
remove rel=help from test references

svgeesus commented on svg/painting/reftests/marker-shape-001.svg:
```diff
@@ -0,0 +1,62 @@
+          href="https://www.w3.org/TR/SVG2/painting.html#Markers"/>
+    <html:link rel="match"  href="marker-shape-001-ref.svg" />
+    <metadata class="flags">TOKENS</metadata>
+    <desc class="assert">TEST ASSERTION</desc>
```
If possible, add an actual test assertion. If not, delete the boilerplate assertion.

svgeesus reviewed (changes requested):
Technical content of the tests looks correct (rects and equivalent paths)

Tavmjong committed:
Remove unneeded test meta data.

ewilligers commented on svg/painting/reftests/marker-shape-001-ref.svg:
```diff
@@ -0,0 +1,49 @@
+  xmlns:xlink="http://www.w3.org/1999/xlink"
+  xmlns:html="http://www.w3.org/1999/xhtml">
+  <g id="testmeta">
+    <title>Hatch: Basic hatch.</title>
```
These tests are not about hatch

Tavmjong committed:
Fix titles.

AmeliaBR reviewed (commented):
Remove the unused font style information.

It shouldn't prevent merging of this PR, but:
There should really be tests for all the basic shapes, and those will  need to be more specific than just comparing against a matching path (because some browsers currently paint too many markers for arcs in paths).

AmeliaBR commented:
See discussion in https://github.com/w3c/svgwg/issues/753 about possibly changing some of the spec behavior, which could affect these tests.

Tavmjong committed:
Remove unused <style> element (text).

Tavmjong committed:
Add reftests for markers on circles and ellipses.

Tavmjong commented:
I've added circle/ellipse tests. And removed unused style element.

Tavmjong committed:
Fix reference rendering in response to resolution that there is no mid-marker at start/end of path equivalent for shapes.
Use straight lines for markers in reference SVGs to avoid rendering error by some browsers (too many markers on curves).
Fixed bad SVG for circle/ellipse test SVGs.
